testhelper/docker/resource/transformer/transformer_backend_config.go

```diff
@@ -0,0 +1,53 @@
+package transformer
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/google/uuid"
+)
+
+const (
+	getByVersionIdEndPoint = "/transformation/getByVersionId"
+	versionIDKey           = "versionId"
+)
+
+type mockHttpServer struct {
+	Transformations map[string]string
+}
+
+func (m *mockHttpServer) handleGetByVersionId(w http.ResponseWriter, r *http.Request) {
+	transformationVersionId := r.URL.Query().Get(versionIDKey)
+	transformationCode, ok := m.Transformations[transformationVersionId]
+	if !ok {
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
+	transformationCode = sanitiseTransformationCode(transformationCode)
+	w.WriteHeader(http.StatusOK)
+	_, err := w.Write([]byte(fmt.Sprintf(`{
+		"id": "%s",
+		"createdAt": "2023-03-27T11:40:00.894Z",
+		"updatedAt": "2023-03-27T11:40:00.894Z",
+		"versionId": "%s",
+		"name": "Add Transformed field",
+		"description": "",
+		"code": "%s",
+		"secretsVersion": null,
+		"codeVersion": "1",
+		"language": "javascript",
+		"imports": [],
+		"secrets": {}
+	}`, uuid.NewString(), transformationVersionId, transformationCode)))
+	if err != nil {
+		return
+	}
+	w.WriteHeader(http.StatusInternalServerError)
+}
+
+func sanitiseTransformationCode(transformationCode string) string {
+	sanitisedTransformationCode := strings.ReplaceAll(transformationCode, "\t", " ")
+	sanitisedTransformationCode = strings.ReplaceAll(sanitisedTransformationCode, "\n", "\\n")
+	return sanitisedTransformationCode
+}
```

Comment:
We don't need a struct, just a producer function which will produce the `httptest.Server`.

```suggestion
func NewTestBackendConfigServer(transformations map[string]string) *kithttptest.Server {
	mux := http.NewServeMux()
	mux.HandleFunc("/transformation/getByVersionId", getByVersionIdHandler(transformations))
	return kithttptest.NewServer(mux)
}

func getByVersionIdHandler(transformations map[string]string) func(http.ResponseWriter, *http.Request) {
	return func(w http.ResponseWriter, r *http.Request) {
		transformationVersionId := r.URL.Query().Get("versionId")
		transformationCode, ok := transformations[transformationVersionId]
		if !ok {
			w.WriteHeader(http.StatusNotFound)
			return
		}
		transformationCode = sanitiseTransformationCode(transformationCode)
		w.WriteHeader(http.StatusOK)
		_, err := w.Write([]byte(fmt.Sprintf(`{
		"id": "%s",
		"createdAt": "2023-03-27T11:40:00.894Z",
		"updatedAt": "2023-03-27T11:40:00.894Z",
		"versionId": "%s",
		"name": "Add Transformed field",
		"description": "",
		"code": "%s",
		"secretsVersion": null,
		"codeVersion": "1",
		"language": "javascript",
		"imports": [],
		"secrets": {}
	}`, uuid.NewString(), transformationVersionId, transformationCode)))
		if err != nil {
			return
		}
		w.WriteHeader(http.StatusInternalServerError)
	}
}

func sanitiseTransformationCode(transformationCode string) string {
	sanitisedTransformationCode := strings.ReplaceAll(transformationCode, "\t", " ")
	sanitisedTransformationCode = strings.ReplaceAll(sanitisedTransformationCode, "\n", "\\n")
	return sanitisedTransformationCode
}
```